pkg/build/builder/common.go

```diff
@@ -81,3 +84,45 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// runScriptInContainer runs a Bash script in an ephemeral Docker container
+// started off imageID, and returns an error if the script cannot be run or
+// returns a non-zero exit code.
+func runScriptInContainer(client DockerClient, script, imageID string) error {
+	// Read CPU and memory limits from /sys/fs/cgroup.
+	// These limits are used to enforce limits on the container that runs
+	// the post commit hook.
+	r := intFromFileReader{}
+	cpuShares := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
+	cpuPeriod := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us")
+	cpuQuota := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
+	memory := r.read("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	if r.err != nil {
+		return r.err
+	}
+
+	return dockerRun(client, docker.CreateContainerOptions{
+		// TODO set Name to a reasonable value
+		// (source-to-image/pkg/docker/docker.go doesn't set the Name!)
```

Comment:
@bparees @mfojtik one more.

If we set `Name: ""`, just like we have in `source-to-image/pkg/docker/docker.go`, the Docker daemon will name the container with random names like "lonely_thompson".

OTOH containers created by Kubernetes have the "k8s_" prefix:
- k8s_POD.b89fddc6_beego-example-1-build_demo_d28a1581-bf86-11e5-b1c0-080027c5bfa9_a0521303
- k8s_docker-build.75576627_beego-example-1-build_demo_d28a1581-bf86-11e5-b1c0-080027c5bfa9_035cc1ac
- k8s_deployment.e0fb8944_beego-example-1-deploy_demo_15217cd9-bf87-11e5-b1c0-080027c5bfa9_29eceb63

Our cleanup scripts rely on the prefix to remove Kube/OpenShift-related containers.

WDYT we should do here? Don't care? Use a Kube helper to generate the name?